src/app-layer-htp.c

```diff
@@ -1630,7 +1630,11 @@ static int HtpRequestBodyHandlePOSTorPUT(HtpState *hstate, HtpTxUserData *htud,
         }
 
         if (filename != NULL) {
-            result = HTPFileOpen(hstate, htud, filename, (uint32_t)filename_len, data, data_len,
+            if (filename_len > UINT16_MAX) {
```

Comment:
should this lead to an event? Also 64k seems very high still. If we consider `PATH_MAX` to be 4k for example.

Reply:
we also have to consider that a 64k name leads to JSON records > 64k

Reply:
Event seems relevant.
`PATH_MAX` seems a better limit as well.
Will do

Reply:
I think I would like us to define our own PATH_MAX(-like) limit, as I think this behaviour in suri should not depend on whatever an OS might define. Since on Linux the default seems 4k I think that might be a good value to pick.